### PR TITLE
Add methods to get proposals

### PR DESCRIPTION
**Pull request by sirnicolaz**
- `getProposals` to get all existing proposals (might break when too many proposals are there)
- `getProposals(uint256 page)` to get paginated proposals

Check method documentation for details on sorting.

### CODE REVIEW

File: evm/contracts/Propcorn.sol [hunk 1]
```diff
@@ -103,11 +106,40 @@ contract Propcorn is Initializable, UUPSUpgradeable, OwnableUpgradeable {
...
+    /**
+     * Returns all proposals, sorted in ascending chronological order
+     * (first one created first).
+     */
+    function getProposals() public view returns (Proposal[] memory) {
+        return proposals;
+    }
+
+    /**
+     * Returns proposals sorted in descending chornological order
```
Comment (vrde) [inline evm/contracts/Propcorn.sol:118]:
```suggestion
     * Returns proposals sorted in descending chronological order
```

File: evm/contracts/Propcorn.sol [hunk 2]
```diff
@@ -121,81 +153,68 @@ contract Propcorn is Initializable, UUPSUpgradeable, OwnableUpgradeable {
...
+        Proposal storage proposal = proposals[index];
 
         _checkFundability(proposal);
 
-        uint256 key = uint256(
-            keccak256(abi.encodePacked(msg.sender, account, index))
-        );
-
-        uint256 toReturn = _addressAndProposalToBalance[key];
+        uint256 toReturn = funderToProposalBalance[msg.sender][index];
```
Comment (vrde) [inline evm/contracts/Propcorn.sol:201]: 😬

**Review by vrde — APPROVED** [review]
Great, it looks much better now.

File: evm/contracts/Propcorn.sol (hunk 2 above)
Comment (sirnicolaz) [reply evm/contracts/Propcorn.sol:201]: in the end that had to change as well, given that the author address is not part of the key anymore

**Pull request merged**